<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>609</width>
    <height>525</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" rowspan="4">
     <widget class="MyPanelOpenGL" name="panelGL" native="true"/>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="circleGroup">
      <property name="title">
       <string>Circle</string>
      </property>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>46</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Radius:</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="doubleSpinBox">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>20</y>
         <width>41</width>
         <height>22</height>
        </rect>
       </property>
       <property name="decimals">
        <number>1</number>
       </property>
       <property name="minimum">
        <double>1.000000000000000</double>
       </property>
       <property name="maximum">
        <double>50.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>46</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Verticies:</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_3">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>60</y>
         <width>42</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <number>5</number>
       </property>
       <property name="maximum">
        <number>60</number>
       </property>
      </widget>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QGroupBox" name="rectangleGroup">
      <property name="title">
       <string>Rectangle</string>
      </property>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>46</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Width:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>46</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Height:</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="doubleSpinBox_2">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>20</y>
         <width>41</width>
         <height>22</height>
        </rect>
       </property>
       <property name="decimals">
        <number>1</number>
       </property>
       <property name="minimum">
        <double>1.000000000000000</double>
       </property>
       <property name="maximum">
        <double>20.000000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="doubleSpinBox_3">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>50</y>
         <width>41</width>
         <height>22</height>
        </rect>
       </property>
       <property name="decimals">
        <number>1</number>
       </property>
       <property name="minimum">
        <double>1.000000000000000</double>
       </property>
       <property name="maximum">
        <double>20.000000000000000</double>
       </property>
      </widget>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QGroupBox" name="triangleGroup">
      <property name="title">
       <string>Triangle</string>
      </property>
      <widget class="QComboBox" name="pointBox">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>69</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maxVisibleItems">
        <number>3</number>
       </property>
       <item>
        <property name="text">
         <string>Point 1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Point 2</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Point 3</string>
        </property>
       </item>
      </widget>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QPushButton" name="quitBtn">
      <property name="maximumSize">
       <size>
        <width>121</width>
        <height>23</height>
       </size>
      </property>
      <property name="text">
       <string>Quit</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>609</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuDraw">
    <property name="title">
     <string>Draw</string>
    </property>
    <addaction name="actionRectangle"/>
    <addaction name="actionCircle"/>
    <addaction name="actionTriangle"/>
    <addaction name="actionCustom"/>
   </widget>
   <addaction name="menuDraw"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionRectangle">
   <property name="text">
    <string>Rectangle</string>
   </property>
  </action>
  <action name="actionCircle">
   <property name="text">
    <string>Circle</string>
   </property>
  </action>
  <action name="actionTriangle">
   <property name="text">
    <string>Triangle</string>
   </property>
  </action>
  <action name="actionCustom">
   <property name="text">
    <string>Custom</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyPanelOpenGL</class>
   <extends>QWidget</extends>
   <header>mypanelopengl.h</header>
   <container>1</container>
   <slots>
    <signal>draw_rect(double,double)</signal>
    <slot>changeSides(int)</slot>
    <slot>changeRadius(double)</slot>
    <slot>changeCircleRadius(double)</slot>
    <slot>changeCircleVerticies(int)</slot>
    <slot>changeRectangleWidth(double)</slot>
    <slot>changeRectangleHeight(double)</slot>
    <slot>setDrawMode(QString)</slot>
    <slot>setTrianglePoint1(QString)</slot>
    <slot>setCursorTriangle(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>quitBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>564</x>
     <y>547</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>actionTriggered(QString)</signal>
   <receiver>panelGL</receiver>
   <slot>setDrawMode(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>259</y>
    </hint>
    <hint type="destinationlabel">
     <x>235</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_2</sender>
   <signal>valueChanged(double)</signal>
   <receiver>panelGL</receiver>
   <slot>changeRectangleWidth(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>554</x>
     <y>285</y>
    </hint>
    <hint type="destinationlabel">
     <x>383</x>
     <y>395</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_3</sender>
   <signal>valueChanged(double)</signal>
   <receiver>panelGL</receiver>
   <slot>changeRectangleHeight(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>563</x>
     <y>309</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>431</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_3</sender>
   <signal>valueChanged(int)</signal>
   <receiver>panelGL</receiver>
   <slot>changeCircleVerticies(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>572</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>456</x>
     <y>237</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>panelGL</receiver>
   <slot>changeCircleRadius(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>559</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>389</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>actionTriggered(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>disableWidgets(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>259</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pointBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>panelGL</receiver>
   <slot>setCursorTriangle(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>510</x>
     <y>460</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>458</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>actionTriggered(QString)</signal>
  <slot>on_actionRectangle_triggered()</slot>
  <slot>on_actionCircle_triggered()</slot>
  <slot>on_actionTriangle_triggered()</slot>
  <slot>on_actionCustom_triggered()</slot>
  <slot>disableWidgets(QString)</slot>
  <slot>openTriangleDialog()</slot>
 </slots>
</ui>
